contracts/staking/ServiceStakingBase.sol

```diff
@@ -508,7 +500,12 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
         }
 
         // Call the checkpoint
-        (uint256[] memory serviceIds, , , , , ) = checkpoint();
+        (uint256[] memory serviceIds, , , , , bool success) = checkpoint();
+
+        // If the checkpoint was not successful, the serviceIds set is not returned and needs to be allocated
+        if (!success) {
+            serviceIds = getServiceIds();
+        }
```

Comment:
Checkpoint now might not return `serviceIds` in memory if it was not successful. So in that case it must be obtained separately.